From 48faf64faafddad1d6126041596311ae5da6dcf9 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sun, 18 Jan 2015 11:18:13 -0800
Subject: [PATCH 2/8] Define EXC_MASK_CRASH and MACH_EXCEPTION_CODES if they're
 not defined in the SDK

The 10.4 SDK does not define these macros

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 58e0ca509dce8b37570b5e15243320fc07a5d6fa)
---
 lib/Support/Unix/Signals.inc | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git llvm_release_39/lib/Support/Unix/Signals.inc macports_release_39/lib/Support/Unix/Signals.inc
index 55fd76d375a..e39586c7f07 100644
--- llvm_release_39/lib/Support/Unix/Signals.inc
+++ macports_release_39/lib/Support/Unix/Signals.inc
@@ -480,6 +480,15 @@ void llvm::sys::PrintStackTraceOnErrorSignal(StringRef Argv0,
   AddSignalHandler(PrintStackTraceSignalHandler, nullptr);
 
 #if defined(__APPLE__) && defined(ENABLE_CRASH_OVERRIDES)
+
+/* These aren't defined in the 10.4 SDK, so provide them here */
+#ifndef EXC_MASK_CRASH
+#define EXC_MASK_CRASH 0x400
+#endif
+#ifndef MACH_EXCEPTION_CODES
+#define MACH_EXCEPTION_CODES 0x80000000
+#endif
+
   // Environment variable to disable any kind of crash dialog.
   if (DisableCrashReporting || getenv("LLVM_DISABLE_CRASH_REPORT")) {
     mach_port_t self = mach_task_self();
-- 
2.11.0

